tools/notification-configuration/common/Helpers/GitHubNameResolver.cs

```diff
@@ -17,6 +18,7 @@ public class GitHubNameResolver : IDisposable
         private readonly ICslQueryProvider client;
         private readonly string kustoTable;
         private readonly ILogger<GitHubNameResolver> logger;
+        private static MemoryCache memoryCache = new MemoryCache(new MemoryCacheOptions());
```

Comment:
What use MemoryCache here? What does it buy us over a simple dictionary for caching values? Are we hitting some sort of in process memory limit? I would be a little surprised if that is the case. If there isn't any of that happening I wouldn't take a dependency on this extra library if a Dictionary will do the job.

Reply:
We're definitely not at risk of exhausting memory. 

We resolve names concurrently so I was looking for something that was thread safe: https://github.com/Azure/azure-sdk-tools/blob/master/tools/notification-configuration/github-codeowner-subscriber/Program.cs#L126

Some implementations of `MemoryCache` are [documented to be thread safe](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.caching.memorycache?view=dotnet-plat-ext-5.0#thread-safety). However, the [version that I ended up pulling in](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.memorycache?view=dotnet-plat-ext-5.0) does not say it's thread safe in the latest documents I find (it also doesn't say it's not thread safe). 

To the end of thread safety and avoiding pulling in additional nuget dependencies, one could use `ConcurrentDictionary`